plugins/woocommerce/includes/class-wc-shortcodes.php

```diff
@@ -523,7 +525,7 @@ public static function product_page( $atts ) {
 		 * @param int   $product_id       Product ID.
 		 * @return array
 		 */
-		$invalid_statuses = apply_filters( 'woocommerce_shortcode_product_page_invalid_statuses', array( 'trash' ), $product_id );
+		$invalid_statuses = apply_filters( 'woocommerce_shortcode_product_page_invalid_statuses', array( OrderStatus::TRASH ), $product_id );
```

Comment:
Here also, I think we're interested in the product (post) status, not the order status.